sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosPartitionPlanner.scala

```diff
@@ -398,7 +408,18 @@ private object CosmosPartitionPlanner extends BasicLoggingTrait {
           case _: ReadMaxRows =>
             val gap = math.max(0, metadata.latestLsn - metadata.startLsn)
             val weightFactor = metadata.getWeightedLsnGap.toDouble / totalWeightedLsnGap.get
-            val allowedRate = (weightFactor * gap).toLong
+            val allowedRate = if ((weightFactor * gap) < 1) {
+              1
+            }
+            else {
```

Comment:
NIT 
Use instead - at least for me easier to read?
(weightFactor * gap).toLong.max(1)

Reply:
That is actually very nice!